src/Microsoft.AspNetCore.Server.IIS/HttpProtocol.cs

```diff
@@ -11,12 +11,13 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.HttpSys.Internal;
 using Microsoft.AspNetCore.WebUtilities;
 using Microsoft.Extensions.Primitives;
 
 namespace Microsoft.AspNetCore.Server.IIS
 {
-    public abstract partial class HttpProtocol : IDisposable
+    internal abstract partial class HttpProtocol : NativeRequestContext, IDisposable
```

Comment:
Inheriting from NativeRequestContext doesn't make sense from an object model perspective. It adds API to HttpProtocol that don't make sense.

Reply:
To quote @davidfowl "but the allocationsssssssssssss"

Reply:
It's ok @Tratcher its internal and saves an allocation, it's good 😄